src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -351,11 +420,37 @@ export class SavedObjectsClient {
     return resolveUrl(API_BASE_URL, join(...path));
   }
 
-  private request({ method, path, query, body }: RequestParams) {
-    if (method === 'GET' && body) {
-      return Promise.reject(new Error('body not permitted for GET requests'));
-    }
-
-    return kfetch({ method, pathname: path, query, body: JSON.stringify(body) });
+  /**
+   * To ensure we don't break backwards compatibility, savedObjectsFetch keeps
+   * the old kfetch error format of `{res: {status: number}}` whereas `http.fetch`
+   * uses `{response: {status: number}}`
+   */
+  private savedObjectsFetch(path: string, { method, query, body }: HttpFetchOptions) {
+    return this.http.fetch(path, { method, query, body }).catch(err => {
+      const kfetchError = Object.assign(err, { res: err.response });
+      return Promise.reject(kfetchError);
+    });
   }
 }
+
+/**
+ * Returns a new object with the own properties of `obj`, but the
+ * keys renamed according to the `keysMap`.
+ *
+ * If a key is not found in `keysMap` it is omitted from the result.
```

Comment:
can it happen that we filter out desirable values? the logic before just renamed keys

Reply:
It's kinda nasty that we have different naming scheme on the browser vs server. The previous implementation made it hard to see what is actually being renamed, so I tried to make it more explicit and strictly typed. Because we have the server vs browser return types typescript guarantees the `pickAndRenameKeys` mapping object correctly maps between the two.

Reply:
shouldn't we at least log a warning if an unknown key is present in a schema?

Reply:
I thought about this again, I wanted to pick only known keys from the `SavedObjectsFindOptions` so that we don't send unknown query parameters to the server, but it doesn't make sense to pick keys from the server response. So I changed this function to `renameKeys` to always trust the server.